<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>1 Introduction | _main.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="1 Introduction | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="1 Introduction | _main.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="biology.html"/>
<script src="libs/header-attrs-2.16/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="index.html#preface">Preface<span></span></a>
<ul>
<li><a href="index.html#author">Author<span></span></a>
<ul>
<li><a href="index.html#became-a-data-scientist">Became a data scientist<span></span></a></li>
<li><a href="index.html#how-did-i-started-programing">How did I started programing?<span></span></a></li>
<li><a href="index.html#more-background">More background<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="2" data-path="biology.html"><a href="biology.html"><i class="fa fa-check"></i><b>2</b> Basic biology<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="biology.html"><a href="biology.html#blood-samples"><i class="fa fa-check"></i><b>2.1</b> Blood samples<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="biology.html"><a href="biology.html#dna"><i class="fa fa-check"></i><b>2.2</b> DNA<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="biology.html"><a href="biology.html#common-terms-in-populational-genetics"><i class="fa fa-check"></i><b>2.2.1</b> Common terms in populational genetics<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="biology.html"><a href="biology.html#rna"><i class="fa fa-check"></i><b>2.3</b> RNA<span></span></a></li>
<li class="chapter" data-level="2.4" data-path="biology.html"><a href="biology.html#protein"><i class="fa fa-check"></i><b>2.4</b> Protein<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cell-type.html"><a href="cell-type.html"><i class="fa fa-check"></i><b>3</b> Cell type<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="cell-type.html"><a href="cell-type.html#flowcytometry-data"><i class="fa fa-check"></i><b>3.1</b> flowcytometry data<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="cell-type.html"><a href="cell-type.html#cell-type-deconvolution"><i class="fa fa-check"></i><b>3.2</b> Cell type deconvolution<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html"><i class="fa fa-check"></i><b>4</b> Bulk RNA seqeuncing analysis<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#preprocessing"><i class="fa fa-check"></i><b>4.1</b> Preprocessing<span></span></a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#filtering"><i class="fa fa-check"></i><b>4.1.1</b> Filtering<span></span></a></li>
<li class="chapter" data-level="4.1.2" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#normalization"><i class="fa fa-check"></i><b>4.1.2</b> Normalization<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#differential-expression-analysis"><i class="fa fa-check"></i><b>4.2</b> Differential expression analysis<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#differential-co-expression-analysis"><i class="fa fa-check"></i><b>4.3</b> Differential co-expression analysis<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#pathway-analysis"><i class="fa fa-check"></i><b>4.4</b> Pathway analysis<span></span></a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>4.4.1</b> Gene set enrichment analysis (GSEA)<span></span></a></li>
<li class="chapter" data-level="4.4.2" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#single-sample-gene-set-enrichment-analysis-ssgsea"><i class="fa fa-check"></i><b>4.4.2</b> Single sample gene set enrichment analysis (ssGSEA)<span></span></a></li>
<li class="chapter" data-level="4.4.3" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#gene-set-variation-analysis-gsva"><i class="fa fa-check"></i><b>4.4.3</b> Gene set variation analysis (GSVA)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#clustering"><i class="fa fa-check"></i><b>4.5</b> Clustering<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Nanostring.html"><a href="Nanostring.html"><i class="fa fa-check"></i><b>5</b> Nanostring technology<span></span></a></li>
<li class="chapter" data-level="6" data-path="proteinomics.html"><a href="proteinomics.html"><i class="fa fa-check"></i><b>6</b> Proteinomics<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="proteinomics.html"><a href="proteinomics.html#olink"><i class="fa fa-check"></i><b>6.1</b> OLINK<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="proteinomics.html"><a href="proteinomics.html#semalogic"><i class="fa fa-check"></i><b>6.2</b> Semalogic<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html"><i class="fa fa-check"></i><b>7</b> Pharmacogenetics analysis<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#basic-genetic-concepts"><i class="fa fa-check"></i><b>7.1</b> Basic genetic concepts<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#pharmacoscan"><i class="fa fa-check"></i><b>7.2</b> Pharmacoscan<span></span></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#software-for-analysis"><i class="fa fa-check"></i><b>7.2.1</b> Software for analysis<span></span></a></li>
<li class="chapter" data-level="7.2.2" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#single-nucleotide-variant-data"><i class="fa fa-check"></i><b>7.2.2</b> Single nucleotide variant data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#hla-region-targeted-sequencing"><i class="fa fa-check"></i><b>7.3</b> HLA region targeted sequencing<span></span></a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#hla-nomenclature-and-sequencing-output"><i class="fa fa-check"></i><b>7.3.1</b> HLA nomenclature and sequencing output<span></span></a></li>
<li class="chapter" data-level="7.3.2" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#drb1-and-drb3drb4drb5"><i class="fa fa-check"></i><b>7.3.2</b> DRB1 and DRB3/DRB4/DRB5<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#reference"><i class="fa fa-check"></i><b>7.4</b> Reference<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="multiplicity.html"><a href="multiplicity.html"><i class="fa fa-check"></i><b>8</b> Multiplicity<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="multiplicity.html"><a href="multiplicity.html#family-wise-comparison"><i class="fa fa-check"></i><b>8.1</b> Family wise comparison<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="multiplicity.html"><a href="multiplicity.html#false-discovery-rate-fdr"><i class="fa fa-check"></i><b>8.2</b> False discovery rate (FDR)<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="multiplicity.html"><a href="multiplicity.html#local-fdr"><i class="fa fa-check"></i><b>8.3</b> Local FDR<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html"><i class="fa fa-check"></i><b>9</b> Power and sample size<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#sample-size-calculation-in-in-vitro-studies"><i class="fa fa-check"></i><b>9.1</b> Sample size calculation in <em>in vitro</em> studies<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#sample-size-calculation-in-omics-studies"><i class="fa fa-check"></i><b>9.2</b> Sample size calculation in <em>omics</em> studies<span></span></a></li>
<li class="chapter" data-level="9.3" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#sample-size-calculation-for-calibration-studies"><i class="fa fa-check"></i><b>9.3</b> Sample size calculation for calibration studies<span></span></a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#agreement-between-two-measurements"><i class="fa fa-check"></i><b>9.3.1</b> 1. Agreement between two measurements<span></span></a></li>
<li class="chapter" data-level="9.3.2" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#sample-size-for-calibrations"><i class="fa fa-check"></i><b>9.3.2</b> 2. Sample size for calibrations<span></span></a></li>
<li class="chapter" data-level="9.3.3" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#choosing-samples-range"><i class="fa fa-check"></i><b>9.3.3</b> Choosing samples range<span></span></a></li>
<li class="chapter" data-level="9.3.4" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#estimating-coefficient-variation-cv-within-the-same-measures"><i class="fa fa-check"></i><b>9.3.4</b> Estimating coefficient variation (CV) within the same measures<span></span></a></li>
<li class="chapter" data-level="9.3.5" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#final-consideration-for-sample-size"><i class="fa fa-check"></i><b>9.3.5</b> Final consideration for sample size<span></span></a></li>
<li class="chapter" data-level="9.3.6" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#reference-1"><i class="fa fa-check"></i><b>9.3.6</b> Reference<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html"><i class="fa fa-check"></i><b>10</b> Detection limit of assays<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#upper-and-lower-detection-limit"><i class="fa fa-check"></i><b>10.1</b> Upper and lower detection limit<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#violation-in-distribution-assumption-and-bias-in-effect-estimate"><i class="fa fa-check"></i><b>10.2</b> Violation in distribution assumption and bias in effect estimate<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#methods-for-detection-limit-problem"><i class="fa fa-check"></i><b>10.3</b> Methods for detection limit problem<span></span></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#substitution"><i class="fa fa-check"></i><b>10.3.1</b> Substitution<span></span></a></li>
<li class="chapter" data-level="10.3.2" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#km"><i class="fa fa-check"></i><b>10.3.2</b> KM<span></span></a></li>
<li class="chapter" data-level="10.3.3" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#ros"><i class="fa fa-check"></i><b>10.3.3</b> ROS<span></span></a></li>
<li class="chapter" data-level="10.3.4" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#mle"><i class="fa fa-check"></i><b>10.3.4</b> MLE<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#comparisons-of-the-methods"><i class="fa fa-check"></i><b>10.4</b> Comparisons of the methods<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="visualization.html"><a href="visualization.html"><i class="fa fa-check"></i><b>11</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="visualization.html"><a href="visualization.html#packages"><i class="fa fa-check"></i><b>11.1</b> packages<span></span></a></li>
<li class="chapter" data-level="11.2" data-path="visualization.html"><a href="visualization.html#basic-static-plot"><i class="fa fa-check"></i><b>11.2</b> Basic static plot<span></span></a></li>
<li class="chapter" data-level="11.3" data-path="visualization.html"><a href="visualization.html#special-plots"><i class="fa fa-check"></i><b>11.3</b> Special plots<span></span></a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="visualization.html"><a href="visualization.html#genetic-analysis"><i class="fa fa-check"></i><b>11.3.1</b> Genetic analysis<span></span></a></li>
<li class="chapter" data-level="11.3.2" data-path="visualization.html"><a href="visualization.html#rna-seq-analysis"><i class="fa fa-check"></i><b>11.3.2</b> RNA seq analysis<span></span></a></li>
<li class="chapter" data-level="11.3.3" data-path="visualization.html"><a href="visualization.html#clustering-analysis"><i class="fa fa-check"></i><b>11.3.3</b> Clustering analysis<span></span></a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="intro" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">1</span> Introduction<a href="intro.html#intro" class="anchor-section" aria-label="Anchor link to header"></a></h1>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="biology.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
