<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Pharmacogenetics analysis | _main.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Pharmacogenetics analysis | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Pharmacogenetics analysis | _main.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="proteinomics.html"/>
<link rel="next" href="multiplicity.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#author"><i class="fa fa-check"></i>Author</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#became-a-data-scientist"><i class="fa fa-check"></i>Became a data scientist</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#how-did-i-started-programing"><i class="fa fa-check"></i>How did I started programing?</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#more-background"><i class="fa fa-check"></i>More background</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="biology.html"><a href="biology.html"><i class="fa fa-check"></i><b>2</b> Basic biology</a><ul>
<li class="chapter" data-level="2.1" data-path="biology.html"><a href="biology.html#blood-samples"><i class="fa fa-check"></i><b>2.1</b> Blood samples</a></li>
<li class="chapter" data-level="2.2" data-path="biology.html"><a href="biology.html#dna"><i class="fa fa-check"></i><b>2.2</b> DNA</a><ul>
<li class="chapter" data-level="2.2.1" data-path="biology.html"><a href="biology.html#common-terms-in-populational-genetics"><i class="fa fa-check"></i><b>2.2.1</b> Common terms in populational genetics</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="biology.html"><a href="biology.html#rna"><i class="fa fa-check"></i><b>2.3</b> RNA</a></li>
<li class="chapter" data-level="2.4" data-path="biology.html"><a href="biology.html#protein"><i class="fa fa-check"></i><b>2.4</b> Protein</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cell-type.html"><a href="cell-type.html"><i class="fa fa-check"></i><b>3</b> Cell type</a><ul>
<li class="chapter" data-level="3.1" data-path="cell-type.html"><a href="cell-type.html#flowcytometry-data"><i class="fa fa-check"></i><b>3.1</b> flowcytometry data</a></li>
<li class="chapter" data-level="3.2" data-path="cell-type.html"><a href="cell-type.html#cell-type-deconvolution"><i class="fa fa-check"></i><b>3.2</b> Cell type deconvolution</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html"><i class="fa fa-check"></i><b>4</b> Bulk RNA seqeuncing analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#preprocessing"><i class="fa fa-check"></i><b>4.1</b> Preprocessing</a><ul>
<li class="chapter" data-level="4.1.1" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#filtering"><i class="fa fa-check"></i><b>4.1.1</b> Filtering</a></li>
<li class="chapter" data-level="4.1.2" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#normalization"><i class="fa fa-check"></i><b>4.1.2</b> Normalization</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#differential-expression-analysis"><i class="fa fa-check"></i><b>4.2</b> Differential expression analysis</a></li>
<li class="chapter" data-level="4.3" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#differential-co-expression-analysis"><i class="fa fa-check"></i><b>4.3</b> Differential co-expression analysis</a></li>
<li class="chapter" data-level="4.4" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#pathway-analysis"><i class="fa fa-check"></i><b>4.4</b> Pathway analysis</a><ul>
<li class="chapter" data-level="4.4.1" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>4.4.1</b> Gene set enrichment analysis (GSEA)</a></li>
<li class="chapter" data-level="4.4.2" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#single-sample-gene-set-enrichment-analysis-ssgsea"><i class="fa fa-check"></i><b>4.4.2</b> Single sample gene set enrichment analysis (ssGSEA)</a></li>
<li class="chapter" data-level="4.4.3" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#gene-set-variation-analysis-gsva"><i class="fa fa-check"></i><b>4.4.3</b> Gene set variation analysis (GSVA)</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="bulk-rna-seqeuncing-analysis.html"><a href="bulk-rna-seqeuncing-analysis.html#clustering"><i class="fa fa-check"></i><b>4.5</b> Clustering</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Nanostring.html"><a href="Nanostring.html"><i class="fa fa-check"></i><b>5</b> Nanostring technology</a></li>
<li class="chapter" data-level="6" data-path="proteinomics.html"><a href="proteinomics.html"><i class="fa fa-check"></i><b>6</b> Proteinomics</a><ul>
<li class="chapter" data-level="6.1" data-path="proteinomics.html"><a href="proteinomics.html#olink"><i class="fa fa-check"></i><b>6.1</b> OLINK</a></li>
<li class="chapter" data-level="6.2" data-path="proteinomics.html"><a href="proteinomics.html#semalogic"><i class="fa fa-check"></i><b>6.2</b> Semalogic</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html"><i class="fa fa-check"></i><b>7</b> Pharmacogenetics analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#basic-genetic-concepts"><i class="fa fa-check"></i><b>7.1</b> Basic genetic concepts</a></li>
<li class="chapter" data-level="7.2" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#pharmacoscan"><i class="fa fa-check"></i><b>7.2</b> Pharmacoscan</a><ul>
<li class="chapter" data-level="7.2.1" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#software-for-analysis"><i class="fa fa-check"></i><b>7.2.1</b> Software for analysis</a></li>
<li class="chapter" data-level="7.2.2" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#single-nucleotide-variant-data"><i class="fa fa-check"></i><b>7.2.2</b> Single nucleotide variant data</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#hla-region-targeted-sequencing"><i class="fa fa-check"></i><b>7.3</b> HLA region targeted sequencing</a><ul>
<li class="chapter" data-level="7.3.1" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#hla-nomenclature-and-sequencing-output"><i class="fa fa-check"></i><b>7.3.1</b> HLA nomenclature and sequencing output</a></li>
<li class="chapter" data-level="7.3.2" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#drb1-and-drb3drb4drb5"><i class="fa fa-check"></i><b>7.3.2</b> DRB1 and DRB3/DRB4/DRB5</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="pharmacogenetics-analysis.html"><a href="pharmacogenetics-analysis.html#reference"><i class="fa fa-check"></i><b>7.4</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="multiplicity.html"><a href="multiplicity.html"><i class="fa fa-check"></i><b>8</b> Multiplicity</a><ul>
<li class="chapter" data-level="8.1" data-path="multiplicity.html"><a href="multiplicity.html#family-wise-comparison"><i class="fa fa-check"></i><b>8.1</b> Family wise comparison</a></li>
<li class="chapter" data-level="8.2" data-path="multiplicity.html"><a href="multiplicity.html#false-discovery-rate-fdr"><i class="fa fa-check"></i><b>8.2</b> False discovery rate (FDR)</a></li>
<li class="chapter" data-level="8.3" data-path="multiplicity.html"><a href="multiplicity.html#local-fdr"><i class="fa fa-check"></i><b>8.3</b> Local FDR</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html"><i class="fa fa-check"></i><b>9</b> Power and sample size</a><ul>
<li class="chapter" data-level="9.1" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#sample-size-calculation-in-in-vitro-studies"><i class="fa fa-check"></i><b>9.1</b> Sample size calculation in <em>in vitro</em> studies</a></li>
<li class="chapter" data-level="9.2" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#sample-size-calculation-in-omics-studies"><i class="fa fa-check"></i><b>9.2</b> Sample size calculation in <em>omics</em> studies</a></li>
<li class="chapter" data-level="9.3" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#sample-size-calculation-for-calibration-studies"><i class="fa fa-check"></i><b>9.3</b> Sample size calculation for calibration studies</a><ul>
<li class="chapter" data-level="9.3.1" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#agreement-between-two-measurements"><i class="fa fa-check"></i><b>9.3.1</b> 1. Agreement between two measurements</a></li>
<li class="chapter" data-level="9.3.2" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#sample-size-for-calibrations"><i class="fa fa-check"></i><b>9.3.2</b> 2. Sample size for calibrations</a></li>
<li class="chapter" data-level="9.3.3" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#choosing-samples-range"><i class="fa fa-check"></i><b>9.3.3</b> Choosing samples range</a></li>
<li class="chapter" data-level="9.3.4" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#estimating-coefficient-variation-cv-within-the-same-measures"><i class="fa fa-check"></i><b>9.3.4</b> Estimating coefficient variation (CV) within the same measures</a></li>
<li class="chapter" data-level="9.3.5" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#final-consideration-for-sample-size"><i class="fa fa-check"></i><b>9.3.5</b> Final consideration for sample size</a></li>
<li class="chapter" data-level="9.3.6" data-path="power-and-sample-size.html"><a href="power-and-sample-size.html#reference-1"><i class="fa fa-check"></i><b>9.3.6</b> Reference</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html"><i class="fa fa-check"></i><b>10</b> Detection limit of assays</a><ul>
<li class="chapter" data-level="10.1" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#upper-and-lower-detection-limit"><i class="fa fa-check"></i><b>10.1</b> Upper and lower detection limit</a></li>
<li class="chapter" data-level="10.2" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#violation-in-distribution-assumption-and-bias-in-effect-estimate"><i class="fa fa-check"></i><b>10.2</b> Violation in distribution assumption and bias in effect estimate</a></li>
<li class="chapter" data-level="10.3" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#methods-for-detection-limit-problem"><i class="fa fa-check"></i><b>10.3</b> Methods for detection limit problem</a><ul>
<li class="chapter" data-level="10.3.1" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#substitution"><i class="fa fa-check"></i><b>10.3.1</b> Substitution</a></li>
<li class="chapter" data-level="10.3.2" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#km"><i class="fa fa-check"></i><b>10.3.2</b> KM</a></li>
<li class="chapter" data-level="10.3.3" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#ros"><i class="fa fa-check"></i><b>10.3.3</b> ROS</a></li>
<li class="chapter" data-level="10.3.4" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#mle"><i class="fa fa-check"></i><b>10.3.4</b> MLE</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="detection-limit-of-assays.html"><a href="detection-limit-of-assays.html#comparisons-of-the-methods"><i class="fa fa-check"></i><b>10.4</b> Comparisons of the methods</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="visualization.html"><a href="visualization.html"><i class="fa fa-check"></i><b>11</b> Visualization</a><ul>
<li class="chapter" data-level="11.1" data-path="visualization.html"><a href="visualization.html#packages"><i class="fa fa-check"></i><b>11.1</b> packages</a></li>
<li class="chapter" data-level="11.2" data-path="visualization.html"><a href="visualization.html#basic-static-plot"><i class="fa fa-check"></i><b>11.2</b> Basic static plot</a></li>
<li class="chapter" data-level="11.3" data-path="visualization.html"><a href="visualization.html#special-plots"><i class="fa fa-check"></i><b>11.3</b> Special plots</a><ul>
<li class="chapter" data-level="11.3.1" data-path="visualization.html"><a href="visualization.html#genetic-analysis"><i class="fa fa-check"></i><b>11.3.1</b> Genetic analysis</a></li>
<li class="chapter" data-level="11.3.2" data-path="visualization.html"><a href="visualization.html#rna-seq-analysis"><i class="fa fa-check"></i><b>11.3.2</b> RNA seq analysis</a></li>
<li class="chapter" data-level="11.3.3" data-path="visualization.html"><a href="visualization.html#clustering-analysis"><i class="fa fa-check"></i><b>11.3.3</b> Clustering analysis</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pharmacogenetics-analysis" class="section level1 hasAnchor">
<h1><span class="header-section-number">7</span> Pharmacogenetics analysis<a href="pharmacogenetics-analysis.html#pharmacogenetics-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="basic-genetic-concepts" class="section level2 hasAnchor">
<h2><span class="header-section-number">7.1</span> Basic genetic concepts<a href="pharmacogenetics-analysis.html#basic-genetic-concepts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>gene</li>
</ul>
</div>
<div id="pharmacoscan" class="section level2 hasAnchor">
<h2><span class="header-section-number">7.2</span> Pharmacoscan<a href="pharmacogenetics-analysis.html#pharmacoscan" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Samples were genotyped using a microarray platform (PharmacoScan™) that targets genetic variants reported to be associated with adverse drug reactions. The array contains more than 18,000 variants (single nucleotide polymorphisms (SNPs), insertions and deletions, copy number variants) for genes that have been associated with adverse drug reactions, including DILI, and are involved in the processes of drug absorption, distribution, metabolism, excretion (ADME genes) and antigen presentation (ie, HLA genes). The panel includes 927 HLA-related variants.</p>
<p>File format for genomewide analysis</p>
<ul>
<li><code>map</code>, <code>ped</code> files</li>
<li><code>vcf</code> files</li>
</ul>
<div id="software-for-analysis" class="section level3 hasAnchor">
<h3><span class="header-section-number">7.2.1</span> Software for analysis<a href="pharmacogenetics-analysis.html#software-for-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>PLINK</li>
<li>R
<ul>
<li>reading vcf files: <a href="https://knausb.github.io/vcfR_documentation/index.html"><code>vcfR</code></a></li>
</ul></li>
</ul>
</div>
<div id="single-nucleotide-variant-data" class="section level3 hasAnchor">
<h3><span class="header-section-number">7.2.2</span> Single nucleotide variant data<a href="pharmacogenetics-analysis.html#single-nucleotide-variant-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
</div>
<div id="hla-region-targeted-sequencing" class="section level2 hasAnchor">
<h2><span class="header-section-number">7.3</span> HLA region targeted sequencing<a href="pharmacogenetics-analysis.html#hla-region-targeted-sequencing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure">
<img src="misc/hla_map_1000px.png" alt="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5610566/" />
<p class="caption"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5610566/" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5610566/</a></p>
</div>
<p>Labcorp Drug Development performed targeted sequencing of HLA region genes. A next-generation sequencing approach based on the GenDX NGSgo MX11-3 kit™ that provides full coverage of Class I and II HLA genes at high resolution, with the exception of DRB4, was used. For Class I HLA genes A, B, and C, exons 1 through 8 were sequenced. For Class II genes DRB1, DRB3, DRB4, DRB5, DQA1, DQB1, DPA1, and DPB1, exons 1 through 6 were sequenced, with the exception of DRB4, where exons 2 and 3 were sequenced. Exons 2 and 3 code for the peptide binding domain of class I HLA, and exon 3 codes for the peptide domain of class II HLA.</p>
<div id="hla-nomenclature-and-sequencing-output" class="section level3 hasAnchor">
<h3><span class="header-section-number">7.3.1</span> HLA nomenclature and sequencing output<a href="pharmacogenetics-analysis.html#hla-nomenclature-and-sequencing-output" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="hla-allele-naming-nomenclature" class="section level4 hasAnchor">
<h4><span class="header-section-number">7.3.1.1</span> HLA allele naming nomenclature<a href="pharmacogenetics-analysis.html#hla-allele-naming-nomenclature" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="misc/hla_allele_naming.jpg" />
<img src="misc/hla_norm.png" alt="http://hla.alleles.org/nomenclature/naming.html" /></p>
<ul>
<li>The suffix ‘L’ is used to indicate an allele which has been shown to have ‘Low’ cell surface expression when compared to normal levels.</li>
<li>The ‘S’ suffix is used to denote an allele specifying a protein which is expressed as a soluble, ‘Secreted’ molecule but is not present on the cell surface.</li>
<li>The ‘C’ suffix is assigned to alleles that produce proteins that are present in the ‘Cytoplasm’ and not on the cell surface.</li>
<li>An ‘A’ suffix indicates an ‘Aberrant’ expression where there is some doubt as to whether a protein is actually expressed.</li>
<li>A ‘Q’ suffix is used when the expression of an allele is ‘Questionable’, given that the mutation seen in the allele has been shown to affect normal expression levels in other alleles.
Alleles that have been shown not to be expressed - ‘Null’ alleles - have been given the suffix ‘N’.</li>
</ul>
</div>
<div id="sequencing-resolution" class="section level4 hasAnchor">
<h4><span class="header-section-number">7.3.1.2</span> Sequencing resolution<a href="pharmacogenetics-analysis.html#sequencing-resolution" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>High resolution
A high-resolution typing result is defined as a set of alleles that encode the same protein sequence for the region of the HLA molecule called the antigen binding site and that exclude alleles that are not expressed as cell-surface proteins. The antigen binding site includes domain 1 and domain 2 of the class I α polypeptides, and domain 1 of the class II α and domain 1 of the class II β polypeptide chains.</li>
<li>Low resolution
A DNA-based typing result at the level of the digits composing the first field in the DNA-based nomenclature. Examples include: A<em>01; A</em>02. If the resolution corresponds to a serologic equivalent, this typing result should also be called low resolution.</li>
<li>Other levels of resolution
If high resolution cannot be obtained or if the laboratory’s agreement with the entity requesting the testing limits the typing efforts to a subset of alleles, the laboratory may report its results at a level of resolution that falls between high resolution and low resolution. Examples are to consider only those alleles expected to be found in the local population or that are designated as common and well defined.3 A third example is typing that assigns a G group designation (eg, A*02:01:01G).</li>
</ul>
</div>
</div>
<div id="drb1-and-drb3drb4drb5" class="section level3 hasAnchor">
<h3><span class="header-section-number">7.3.2</span> DRB1 and DRB3/DRB4/DRB5<a href="pharmacogenetics-analysis.html#drb1-and-drb3drb4drb5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="misc/drb_arch.png" /></p>
</div>
</div>
<div id="reference" class="section level2 hasAnchor">
<h2><span class="header-section-number">7.4</span> Reference<a href="pharmacogenetics-analysis.html#reference" class="anchor-section" aria-label="Anchor link to header"></a></h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="proteinomics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="multiplicity.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
